tools/clang/lib/SPIRV/DeclResultIdMapper.cpp

```diff
@@ -2668,9 +2688,61 @@ bool DeclResultIdMapper::decorateResourceBindings() {
     }
   }
 
+  decorateResourceHeapsBindings(bindingSet);
   return true;
 }
 
+void DeclResultIdMapper::decorateResourceHeapsBindings(BindingSet &bindingSet) {
+  bool hasResource = false;
+  bool hasSamplers = false;
+  bool hasCounters = false;
```

Comment:
There is more than 1 variable per heap: there is one variable per heap per type:
Texture2D + Texture3D -> 2 variables sharing the same binding.
I could add a vector and store the pointers, but do we really have that many resources to avoid iterating on those twice?

Reply:
Right, I forgot about that. This should be fine then. It might turn out to be a wash. What you currently have might be the simplest code. I'll leave it for you to decide what you want to do.